### ISSUE TITLE
[Have a good start] : connection between back and front

### ISSUE THREAD

**Issue by GN13008** [llucasspot/boomboom-front-react-native#10]
Hi, I am trying to connect back end with front end on IOS Expo.

My back is working I mean i have a success response to /auth/spotify.

On my front when i click on "sign in with spotify" I think my WebView do not go to the spotify view ...

I have an Error loading page (see screen in the next comment)

Perhaps we can improve the readme files to init project ?

Thank's :)

**Comment by GN13008** [comment]
![image](https://github.com/llucasspot/boomboom-front-react-native/assets/81300850/07a6cbe7-1a8a-4fc9-bf6d-5e0407e5f0d6)

![image](https://github.com/llucasspot/boomboom-front-react-native/assets/81300850/0c0438b3-b6fb-4e72-a78b-a09117d73f0b)

![image](https://github.com/llucasspot/boomboom-front-react-native/assets/81300850/c41a17b0-9042-47c3-a9ef-cd37675dfd30)

**Comment by llucasspot** [comment]
Hi GN, what are yours front env variables ? for exemple EXPO_PUBLIC_API_URL is like http://192.168.X.X:3333/api

**Comment by GN13008** [comment]
Front : 
EXPO_PUBLIC_API_URL=http://localhost:3333/api
EXPO_PUBLIC_BY_PASS_SIGN_IN_SCREEN=false
EXPO_PUBLIC_MOCK_MODE=false

**Comment by llucasspot** [comment]
you can not use `localhost` since the app runs on your phone, you localhost will be your iphone but the backend server is on your computer. you need to use your pc's ip ;) 

run `yarn start -c` with `-c` params to clear cache if it is need to reload env vars, i'm not sure

**Comment by GN13008** [comment]
OMG ... that's seems so obvious now ... thank's

I think it will work now ! I just have to create a spotify account because using google authentification seems to be more complexe

**Comment by llucasspot** [comment]
No worries
What do you mean, because you have a spotify account that use google auth to authenticate ? it does not work well if we are in taht case ?

**Comment by GN13008** [comment]
Yes, I think i have some params to do on my google account to accept works with http (not s) or something like that. But no problem I have an other account

Do you have some important args in your back .env ? except secret key and db connection ?
What about params in spotify dev ?

maybe we can rename this issue "have a good start" ^^

**Comment by llucasspot** [comment]
i have `APP_KEY` &  the database connection envs & `SPOTIFY_CLIENT_ID` & `SPOTIFY_CLIENT_SECRET`

in the spotify dev dashboard i have set `Redirect URIs` to `http://192.168.X.X:3333/api/auth/spotify/callback`
and i have added an user email in `User Management`

**Comment by GN13008** [comment]
Okay Thank's, you do not have any URI in your .env (back) ? 

I have a INVALID_CLIENT: Invalide redirect URI

**Comment by llucasspot** [comment]
you can set it with SPOTIFY_API_URL

**Comment by GN13008** [comment]
Ok sign in with spotify works :) 

Thank's now I can work on some Front PR ! 

PS : my .env back 
`# SPOTIFY_API_URL=http://localhost:3333/api/aut
SPOTIFY_CALLBACK_URL=http://192.168.X.X:3333/api/auth/spotify/callback`

Do you have an inventory of what is working and what is not ? (I am still talking about connection between front and back)

**Comment by llucasspot** [comment]
cool ;) 

We did not test the api after the signin, the rest of the front has been dev with EXPO_PUBLIC_MOCK_MODE enabled.

I spoked with @Doelia, he will be managing the frontend part and me the backend part.

I think that the next step that would be great is to better handle form sending and error handling in the frontend.

**Comment by llucasspot** [comment]
I have done some big changes (addint eslint, prettier, etc), i have added the env BASE_API_URL=http://192.168.X.X:3333/api to not need to set the CALLBACK env variables.

please pull the new changes ;)
